extensions_sql_server_virtual_machines.tf

```diff
@@ -0,0 +1,75 @@
+locals {
+  log_analytics_policies = {
+    mdc-log-analytics-arc1-autoprovisioning = {
+      definition_display_name = "[Preview]: Configure Azure Arc-enabled Windows machines with Log Analytics agents connected to default Log Analytics workspace"
+    }
+    mdc-log-analytics-arc2-autoprovisioning = {
+      definition_display_name = "[Preview]: Configure Azure Arc-enabled Linux machines with Log Analytics agents connected to default Log Analytics workspace"
+    }
+  }
+  log_analytics_roles = {
+    role-1 = {
+      name   = "Contributor"
+      policy = "mdc-log-analytics-arc1-autoprovisioning"
+    }
+    role-2 = {
+      name   = "Contributor"
+      policy = "mdc-log-analytics-arc2-autoprovisioning"
+    }
+  }
+}
+
+# Enabling extension - Log Analytics for arc
+data "azurerm_policy_definition" "la_policies" {
+  for_each = (contains(var.mdc_plans_list, "SqlServerVirtualMachines") || contains(var.mdc_plans_list, "Databases")) && !contains(var.mdc_plans_list, "VirtualMachines") ? local.log_analytics_policies : {}
```

Comment:
I would recommend an extra pair of parenthesis:

```hcl
for_each = (contains(var.mdc_plans_list, "SqlServerVirtualMachines") || （contains(var.mdc_plans_list, "Databases")) && !contains(var.mdc_plans_list, "VirtualMachines")） ? local.log_analytics_policies : {}
```

Since this expression has been repeated more than three times, could we extract it into a seperate `local` expression to reduce the duplicated code? Like:

```hcl
locals
{
    create_sql_server_virtual_machines = (contains(var.mdc_plans_list, "SqlServerVirtualMachines") || （contains(var.mdc_plans_list, "Databases")) && !contains(var.mdc_plans_list, "VirtualMachines")）
}